MEGA/Main/Main.ino

```diff
@@ -121,17 +121,88 @@ State state();
 #endif
 
 #if ENABLE_SENSORS
+#define RGB_1_COLOUROUT A5
+#define RGB_1_S2 A3
+#define RGB_1_S3 A4
+#define RGB_2_COLOUROUT A10
+#define RGB_2_S2 A8
+#define RGB_2_S3 A9 
+
   class Sensors {
+    /* Sensor Type : RGB sensors 
+       Sensor Model : XC3708 
+       Number of sensors : 2 (Front and Rear of the carriage) */
+
+    private: 
+      int baudRate = 9600;
+
     public:
+    
       Sensors() {
         // Use ::state to access the state class in this scope
+        Serial.begin(baudRate);
+        pinMode(RGB_1_COLOUROUT,INPUT);
+        pinMode(RGB_1_S2,OUTPUT);
+        pinMode(RGB_1_S3,OUTPUT);
+
+        /*  When RGB_2 sensors are enabled,
+        ie: When carriage's direction is reversed 
+        the sensors at the current Front section of
+        the carriage will be enabled 
+
+        pinMode(RGB_2_COLOUROUT,INPUT); 
+        pinMode(RGB_2_S2,OUTPUT);
+        pinMode(RGB_2_S3,OUTPUT); */
+
       }
 
       /*
         Check color sensors sense correct colors, and other stuff
       */
       void calibrate() {
-        
+        int intensity = getIntensity();
+
+        // Detecting the percentage of Red colour 
+        detectRed();
+        Serial.print("RED :");
+        Serial.print(getIntensity());
+        Serial.print("% ");
+
+        // Detecting the percentage of Green colour 
+        detectGreen();
+        Serial.print("GREEN :");
+        Serial.print(getIntensity());
+        Serial.print("% ");        
+      }
+
+      /*Checks for Red in the colour detected */
+      void detectRed(){
+        digitalWrite(RGB_1_S2,LOW);
+        digitalWrite(RGB_1_S3,LOW);
+      }
+
+      /*Checks for Green in the colour detected */
+      void detectGreen(){ 
+        digitalWrite(RGB_1_S2,HIGH);
+        digitalWrite(RGB_1_S3,HIGH);
+      }
+
+      //measure intensity with oversampling
+      int getIntensity(){ 
```

Comment:
Might be good to fix up the syntax here to make it a bit easier to read, something like this:
```suggestion
    int getIntensity() { 
        int a=0;
        int b=255;
        for(int i = 0; i < 10; i++) {
            a = a + pulseIn(RGB_1_COLOUROUT,LOW);
        }
        if (a > 9) {
            b = 2550/a;
        }
        return b;
    }
```

Reply:
Yes. Thank you !